### ISSUE TITLE
[Feature Request] Add cbz/cbr formats for conversion to and from pdf/epub/mobi

### ISSUE THREAD

**Issue by applesoff** [Stirling-Tools/Stirling-PDF#775]
I would like to convert my cbz files to epub/pdf and them to mobi so that they can be added to kindle for reading.
Currently cbz/cbr formats are not supported.

**Comment by sbplat** [comment]
Is `ebook-convert` able to convert `cbr`/`cbz` formats?

**Comment by Frooodle** [comment]
Calibre says it supports cbr and cbz so just needs to be added to supported extensions in the list etc

**Comment by applesoff** [comment]
Do I add these myself?

**Comment by sbplat** [comment]
You are welcome to add them!

**Comment by applesoff** [comment]
How does one do that? I am fairly tech illiterate. i couldn't find anything in the github/website on how to do that

**Comment by jdaviescoates** [comment]
yeah, I just went to convert a PDF to an epub but noticed that feature is missing

**Comment by balazs-szucs** [comment]
Hi,

I'd like to work on CBZ. As for the other, EPUB conversion is being worked on here: #3261.

As for CBR; it is based on RAR which is a proprietary format, basically means it won't be possible without a dependency and there are currently no good Java libraries (at least that I found; I might be stand corrected if someones find one, well besides calibre) under permissive license that could handle RAR (which is quite shocking imho but what can you do); it is possible but would probably require some binary e.g., 7zip and some Java wrapper around it. I am open to suggestions on this front.

As for MOBI; the format is complicated by common DRM and other quirks, making it a challenging and a huge headache so probably not something that we would want to undertake.

Imho, CBZ should be very straightforward, I expect to be done in a few days.

I would like to link it to close this issue, since as I said CBR/MOBI, are not really feasible, CBZ will be done by me, EPUB already has a dedicated ticket.

PS. if someone wants some special thing regarding cbz feel free to let me know.

**Comment by balazs-szucs** [comment]
Hi,

I might have jumped the gun on CBR a bit too early. Sorry about that.

**Comment by balazs-szucs** [comment]
Hi, @Frooodle 

Sorry for the ping. For CBR the following library is needed: https://github.com/junrar/junrar

size: approx with everything 180 KB
license: it has permissive license, but due to the fact that RAR is proprietary it has license other than the "big ones"

It is made by same guy is doing the [NightMonkeys](https://github.com/gotson/NightMonkeys) so its pretty good

feel free to let me know if this acceptable or not.

**Comment by Frooodle** [comment]
I believe that would be acceptable 
I'll check again later but for now I'll say yes! Should be fine
